### DISCUSSION TITLE
Allow for changing responsive reading of Psalms by half verse and whole verse

### DISCUSSION THREAD

**Discussion by perryholden** [blocher/dailyoffice2019#150] [Ideas]
@blocher I'd like to add the ability to change the "Psalm Recitation Style" both for Morning and Evening Prayer as well as the "Day's Readings". Since the API calls for MP/EP are based on settings, it'd be fairly easy to add "Psalm Recitation Style" as a setting. From what I can see, there would be no other changes required to the Vue templates.

As for the "Day's Readings", which isn't based on stored settings at all, I think I'd suggest one of two approaches:

1. Add both `whole_verse` and `half_verse` to the `readings` API response. The API response would be much larger (as you'd need 4 variations: "whole 30", "whole 60", "half 30", and "half 60"), but that may not be too big of a deal. We could then add a toggle switch (similar to the 30 Day Cycle / 60 Day Cycle) to toggle between these.
2. Another idea (but more complex) would be to add a `psalm_style=plain` (in addition to the existing `whole_verse`, `half_verse`, and `unison`). This would not return verses with `<strong>` tags but would return somewhat "plain" text, and the Vue frontend would add the relevant tags or font-weight based on the status of the toggle switch.

I used to be more favorable to option 2, but the more I think about it, I am leaning towards option 1. But I wanted to open this up for discussion to get your thoughts. I welcome your feedback. Let me know if I have misunderstood anything.

**Comment by perryholden** [comment]
@blocher - I realized that the "Psalm Recitation Style" is listed as an "Advanced Setting", so it's already there in the full Morning and Evening Prayer. My bad. Also, I see that when set in settings, it also updates it on the daily readings too, which is exactly what I was looking for.

I guess the only thing is that I didn't realize that there even were advanced settings until I dug into the code a bit. I overlooked the "Frequently Used / All Settings" toggle on the bottom of the settings page. I wonder if there is a way to make the advanced settings more visible, as they might be overlooked unless you know to switch the toggle?

**Reply by blocher** [reply]
I think people missing the advanced settings has been an ongoing problem. I am open to any ideas there. I am trying to balance not making the settings page overwhelming with loads of options and allowing full customization. I am open to small tweaks or a full design redo. 

As far as an extra toggle on the readings page, I think I am ok with that. I do not want any toggles in Morning or Evening Prayer, however, as the idea is to not be distracted during prayer.

**Comment by perryholden** [comment]
@blocher - Good news. I have worked the "Psalm Recitation Style" into the Readings section, using an HTML select like the "Translation" and "Psalter Version". Once you approve PR #148, I will submit another PR for this addition. It functions identically to the other HTML selects, including saving in local storage. It does lengthen that section a bit, but it does look good on mobile. Thanks!

<img width="511" alt="The Daily Office 2024-02-17 18-46-50" src="https://github.com/blocher/dailyoffice2019/assets/13052746/78425555-19c1-4c85-a3f2-5750b63573a9">

**Reply by blocher** [reply]
@perryholden Looks great to me!
